python/ray/util/rpdb.py

```diff
@@ -338,16 +340,28 @@ def _post_mortem():
 
 
 def _connect_pdb_client(host, port):
+    if sys.platform == "win32":
+        import msvcrt
     s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     s.connect((host, port))
 
     while True:
         # Get the list of sockets which are readable.
-        read_sockets, write_sockets, error_sockets = select.select(
-            [sys.stdin, s], [], []
-        )
+        if sys.platform == "win32":
+            ready_to_read = select.select([s], [], [], 1)[0]
+            if msvcrt.kbhit():
+                ready_to_read.append(sys.stdin)
+            if not ready_to_read and not sys.stdin.isatty():
+                # in tests, when using pexpect, the pipe makes
+                # the msvcrt.kbhit() trick fail. Assume we are waiting
+                # for stdin, since this will block waiting for input
+                ready_to_read.append(sys.stdin)
```

Comment:
This was annoying to debug. In the test, pexpect uses a PIPE to communicate over `sys.stdin` which does not work correctly with `msvcrt.kbhit()`. "Normal" `ray debug` use will not enter this clause, `sys.stdin.isatty()` will be `True`